cunumeric/array.py

```diff
@@ -3119,16 +3119,25 @@ def _normalize_summation(
                 divisor = reduce(lambda x, y: x * y, self.shape, 1) - ddof
         else:
             if where is not None:
-                divisor = (
-                    where.sum(axis=axis, dtype=dtype, keepdims=keepdims) - ddof
+                divisor = where.sum(
+                    axis=axis, dtype=sum_array.dtype, keepdims=keepdims
                 )
+                if ddof != 0 and not np.isscalar(divisor):
+                    mask = divisor != 0
+                    values = divisor - ddof
+                    divisor._thunk.putmask(mask._thunk, values._thunk)
+                else:
+                    divisor -= ddof
             else:
                 divisor = self.shape[axis] - ddof
 
         # Divide by the number of things in the collapsed dimensions
         # Pick the right kinds of division based on the dtype
-        if np.ndim(divisor) == 0:
+        if isinstance(divisor, ndarray):
+            divisor = divisor.astype(sum_array.dtype)
+        else:
             divisor = np.array(divisor, dtype=sum_array.dtype)  # type: ignore [assignment] # noqa
+
         if dtype.kind == "f" or dtype.kind == "c":
```

Comment:
I think you can just get rid of the `dtype` parameter at this point, and just use `sum_array.dtype` everywhere in this function.